.github/workflows/review.yml

```diff
@@ -43,3 +43,26 @@ jobs:
       run: mv .github/tools/.checkpatch.conf .
     - name: Run checkpatch review
       uses: webispy/checkpatch-action@v9
+  upstream_tests:
+     runs-on: self-hosted
+     timeout-minutes: 120
+     name: upstream tests
+     steps:
+     - uses: actions/checkout@v4
+       with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+     - name: 'Prepare machine'
+       run: |
+          vagrant snapshot restore clean_vm_v1
```

Comment:
although I believe this works I think you need to add "cd .." here (unless its above somewhere and I missed it.  it's best to run vagrant commands in the dir where Vagrantfile lives and by default that is one level up from where each step starts on the runner